src/sinks/nats.rs

```diff
@@ -213,7 +213,12 @@ impl StreamSink<Event> for NatsSink {
                 continue;
             }
 
-            match self.connection.publish(&subject, &bytes).await {
+            let message_size = bytes.len();
+            match self
+                .connection
+                .publish(subject.clone(), bytes.freeze())
+                .await
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏We should take the result of flushing into consideration and if that fails, error the event.

```suggestion
                .publish(subject.clone(), bytes.freeze())
                .map_err(Into::into)
                .and_then(|_| self.connection.flush().map_err(Into::into))
                .await
```

You'll also need to update `NatsEventSendError::error` to be `async_nats::Error` which encompasses the error from both calls.

Reply:
Do you still want the error type to be `async_nats::Error`?